client/js/yt-music/scroller.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * figures out which section the user is currently scrolled to
+ * in youtube-genre.html
+ *
+ * @returns {scroll}
+ */
+function scroller() {
```

Comment:
this function acts like a constructor / class; it might be easier to understand if you convert it to a class instead of attaching properties to the `scroll` object and returning that